test/extended/router/metrics.go

```diff
@@ -306,3 +332,73 @@ func getAuthenticatedURLViaPod(ns, execPodName, url, user, pass string) (string,
 	}
 	return output, nil
 }
+
+func getBearerTokenURLViaPod(ns, execPodName, url, bearer string) (string, error) {
+	cmd := fmt.Sprintf("curl -s -k -H 'Authorization: Bearer %s' %q", bearer, url)
+	output, err := e2e.RunHostCmd(ns, execPodName, cmd)
+	if err != nil {
+		return "", fmt.Errorf("host command failed: %v\n%s", err, output)
+	}
+	return output, nil
+}
+
+func findEnvVarSecret(vars []corev1.EnvVar, key string) (string, string) {
+	for _, v := range vars {
+		if v.Name == key {
+			if v.ValueFrom != nil && v.ValueFrom.SecretKeyRef != nil {
+				ref := v.ValueFrom.SecretKeyRef
+				return ref.Name, ref.Key
+			}
+		}
+	}
+	return "", ""
+}
+
+func findStatsUsernameAndPassword(oc *exutil.CLI, ns string, env []corev1.EnvVar) (string, string, error) {
+	secretName, userKey := findEnvVarSecret(env, "STATS_USERNAME")
+	_, passKey := findEnvVarSecret(env, "STATS_PASSWORD")
+
+	if len(secretName) == 0 || len(userKey) == 0 || len(passKey) == 0 {
+		return "", "", fmt.Errorf("stats username and password not found, env: %v", env)
+	}
+
+	secret, err := oc.AdminKubeClient().Core().Secrets(ns).Get(secretName, metav1.GetOptions{})
+	if err != nil {
+		return "", "", err
+	}
+
+	username, ok1 := secret.Data[userKey]
+	password, ok2 := secret.Data[passKey]
+	if !ok1 || !ok2 {
+		return "", "", fmt.Errorf("secret '%s/%s' does not contain key %q or %q", ns, secretName, userKey, passKey)
+	}
+	return string(username), string(password), nil
+}
+
+func findMetricsBearerToken(oc *exutil.CLI) (string, error) {
```

Comment:
for future discussion: what about running the test pod using an SA with the right permissions?